Prevalence patient search

davidbgomes opened the pull request:
### :pushpin: References

-   **Issue:** Closes#8694afm42 [Patient search in Prevalence module](https://app.clickup.com/t/8694afm42)

### :memo: Implementation

- Add Prevelance patient search logic. Since the trackedEntity endpoint doesn't support the rootJunction=OR, in order to filter by one field or the other, I've created two separate calls for each and joined them.

### :video_camera: Screenshots/Screen capture
<img width="1121" alt="image" src="https://github.com/EyeSeeTea/amr-surveys/assets/44171664/8da0f24b-7702-4201-b517-8280e6327497">


### :fire: Notes to the tester

9sneha-n reviewed (changes requested):
@davidbgomes If i type "aa" in the search and then delete one char "a" then there are duplicate surveys displayed as seen in screenshot. There should be only 2 surveys, but they are displayed twice 
<img width="1728" alt="Screenshot 2024-05-07 at 11 47 49 AM" src="https://github.com/EyeSeeTea/amr-surveys/assets/83749675/ee7fb054-f5a2-4ef0-9e89-469a2d40f0bc">

9sneha-n commented:
@davidbgomes based on slack discussion with miquel on amr-surveys channel, we can use api filter correct?
`/api/tracker/trackedEntities?program=i0msBbbQxYC&orgUnit=di7FkWFGyuS&fields=trackedEntity,attributes&filter=tlRPoWumrSa%3Aeq%3AdAGk9nbjMeI&filter=yEkJlUFeJdP%3Alike%3Aaa&rootJunction=AND`
Let me know if you see any issue with this approach.

9sneha-n commented on src/webapp/components/survey-list/SurveyList.tsx:
```diff
@@ -82,10 +82,18 @@ export const SurveyList: React.FC<SurveyListProps> = ({ surveyFormType }) => {
                         ) && (
                             <ButtonWrapper>
-                                {surveyFormType === "PPSPatientRegister" && (
+                                {["PPSPatientRegister", "PrevalenceCaseReportForm"].includes(
```
could you please create an array  like SURVEYS_WITH_FILTER_ENABLED or something similar and use it everywhere.

davidbgomes commented:
Hi @9sneha-n , addressed the comment about using an array for survey types. As we talked on slack, the endpoints /events and /trackedEntities endpoints no longer work with `rootJunction=OR`, it acts like the default AND. I'll keep it as it for now and we'll see what can be done in that regard.

9sneha-n commented:
Fixed in https://github.com/EyeSeeTea/amr-surveys/pull/59
